### PR TITLE
CLOUDP-259895: Audit conversion tests + method refactor 

### PR DESCRIPTION
**Pull request by filipcirtog**
### All Submissions:

* [ ] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [ ] Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if there is one).
* [ ] Update docs/release-notes/release-notes-template.md if your changes should be included in the release notes for the next release.

### CODE REVIEW

File: internal/translation/audit/conversion_test.go [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+	}
+	for _, tc := range testCases {
+		t.Run(tc.title, func(t *testing.T) {
+			actualResult := NewAuditConfig(tc.input)
+			assert.Equal(t, tc.expectedOutput, actualResult)
+		})
+	}
+}
+
+func TestConvesrion(t *testing.T) {
```
Comment (josvazg) [inline internal/translation/audit/conversion_test.go:103]: Small typo:

```suggestion
func TestConversion(t *testing.T) {
```

**Review by josvazg — APPROVED** [review]
LGTM! great job!

**Comment by s-urbaniak** [comment]
@filipcirtog @josvazg looks very cool indeed, however as in the other PR: please create a jira ticket and link it here.

**Comment by josvazg** [comment]
> @filipcirtog @josvazg looks very cool indeed, however as in the other PR: please create a jira ticket and link it here.

Filip created CLOUDP-259895 already, we just need to link it.

**Comment by helderjs** [comment]
I have generated IP AcessList mock with mockery, and I was in mind to do the same with the existing translation stuff.
Reasons: We don't maintain mocks manually and it is consistent with SDK mocks. With that we handle mocks in the same way everywhere in our unit tests
what do you think @filipcirtog @josvazg @s-urbaniak

**Comment by josvazg** [comment]
> I have generated IP AcessList mock with mockery, and I was in mind to do the same with the existing translation stuff. Reasons: We don't maintain mocks manually and it is consistent with SDK mocks. With that we handle mocks in the same way everywhere in our unit tests what do you think @filipcirtog @josvazg @s-urbaniak ?

Agreed. I will refactor my mocks like that, anything already merged I mean. The dbusers one will already go with mockery.

I suggest we later add automation to generate the mocks from the Makefile, instead of manually with the tool on our own.

**Pull request merged**